portability-worker/src/main/java/org/dataportabilityproject/worker/WorkerMain.java

```diff
@@ -18,17 +18,66 @@
 import com.google.common.util.concurrent.UncaughtExceptionHandlers;
 import com.google.inject.Guice;
 import com.google.inject.Injector;
+import org.dataportabilityproject.api.launcher.ExtensionContext;
+import org.dataportabilityproject.api.launcher.Logger;
+import org.dataportabilityproject.spi.transfer.extension.TransferExtension;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.ServiceLoader;
 
 /**
  * Main class to bootstrap a portabilty worker that will operate on a single job whose state
  * is held in WorkerJobMetadata.
  */
 public class WorkerMain {
-  public static void main(String[] args) throws Exception {
-    Thread.setDefaultUncaughtExceptionHandler(UncaughtExceptionHandlers.systemExit());
-    Injector injector = Guice.createInjector(new WorkerModule());
-    WorkerImpl worker = injector.getInstance(WorkerImpl.class);
-    worker.processJob();
-    System.exit(0);
-  }
+    public static void main(String[] args) throws Exception {
+        Thread.setDefaultUncaughtExceptionHandler(UncaughtExceptionHandlers.systemExit());
+
+        ExtensionContext extensionContext = getExtensionContext();
+
+        List<TransferExtension> extensions = new ArrayList<>();
```

Comment:
Just a thought, we might want to limit it to two transfer extensions, one for import and one for export, for now and eventually one when we head down to separate importer and export processes, just one.

I think in this case, it helps to be better defined and less general.

Reply:
We could do that. One thing I was thinking is the embedded case where multiple extensions are loaded in memory. We could launch multiple workers for that case.